### PR TITLE
Loading history on mobile network

### LINKED ISSUE #7288: Request chat history on demand while on cellular connection
## Pivotal link
https://www.pivotaltracker.com/story/show/163186553

# Problem

Currently, chats' history is requested from the mailservers regardless of what kind of internet connection is used on the device. For cellular connection, it might cause unacceptable traffic consumption. 

# Implementation

When network connection type changes to cellular a dialog with the next options can be shown to the user:
1. stop chats' syncing, continue on demand/when switch back to a different type of connection
2. continue syncing
3. always stop syncing when such event occurs
4. always continue syncing when such even occurs

On `1.` and `2.` a chosen action is applied and the same dialog is shown on next switching to the cellular network.

If the user chooses `3.` or `4.` a chosen action is applied and on the next occurrence the dialog "stop syncing while on cellular"/"continue syncing while on cellular" with two options can be shown:
- ok
- go to settings

In order to not annoy the user with all these dialogs, it should be shown once per session (or once per some period of time). If we have some connection specific badge it might be shown to the user only once at all. But we still have to make settings easily accessible from almost any screen then (as well as that force-requesting button).

# Acceptance Criteria

- when a device switches to cellular network the user can stop syncing
- when a device switches to wifi network syncing is resumed
- when syncing is stopped the user can demand history by tapping some button
- network connection settings are accessible in profile settings

# Notes

From UI/UX side we need:
1. "force-requesting" button
2. connection state badges

### PR DESCRIPTION
**Pull request by rasom**
fix #7288

notes:
1. A bottom sheet is shown when switch from wifi to cellular network
2. The options, selected in the bottom sheet can be changed in profile settings
3. If "remember my choice" was selected next time bottom sheet will not be shown on switching to cellular
4. If syncing on mobile wasn't enabled history messages from mail servers will not be requested. Connection to mail server can still be established.
5. When devices switches back to wifi and app is running, history is requested
6. History is also requested when user decides to enable syncing while app is on cellular network
7. A separate button for force requesting of history (without enabling syncing on mobile network) will be implemented as a separate PR

status: ready

### CODE REVIEW

File: resources/icons/cancel.svg [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (yenda) [inline resources/icons/cancel.svg]: this is already in develop :main-icons/cancel

File: resources/icons/network24.svg [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (yenda) [inline resources/icons/network24.svg]: this is already in develop `main-icons/network`

File: resources/icons/cancel.svg (hunk 1 above)
Comment (rasom) [reply resources/icons/cancel.svg]: that's part of another PR, fixed here https://github.com/status-im/status-react/pull/7396/files

File: src/status_im/ui/screens/mobile_network_settings/events.cljs [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+            (not= :create-account (:view-id db)))
+
+       [(bottom-sheet/show-bottom-sheet
+         {:view :mobile-network})
+        (sheet-defaults)]
+
+       logged-in?
+       [(bottom-sheet/hide-bottom-sheet)]))))
+
+(defn apply-settings
```
Comment (cammellos) [inline src/status_im/ui/screens/mobile_network_settings/events.cljs:37]: Probably it would be good to add some tests for this namespace, also I generally prefer not to use `trooleans` (nil, true, false), at which point I'd rather use strings to capture state

**Review by cammellos — APPROVED** [review]

File: src/status_im/ui/screens/mobile_network_settings/events.cljs (hunk 3 above)
Comment (rasom) [reply src/status_im/ui/screens/mobile_network_settings/events.cljs:37]: do you mean for `syncing-on-mobile-network?` and `remember-syncing-choice?`? These fields are not nillable

Comment (cammellos) [reply src/status_im/ui/screens/mobile_network_settings/events.cljs:37]: `remember?` and there's another example I think in a different file `new-show?` where we explicitly check that is `false?`, although in that  case looks we have less control as coming from props

File: src/status_im/mailserver/core.cljs [hunk 4]
```diff
@@ -332,24 +332,29 @@
...
-           :mailserver/request-messages {:web3     web3
-                                         :mailserver    mailserver
-                                         :request request}}
-          {:db (dissoc db
-                       :mailserver/pending-requests
-                       :mailserver/request-to)})))))
+  (let [network (:network/type db)
+        {:keys [syncing-on-mobile-network?]} (:account/account db)]
+    (when (and
+           (not (and (= network "cellular")
```
Comment (mandrigin) [inline src/status_im/mailserver/core.cljs]: You've defined a function called `cellular?` Maybe we can re-use it  instead of keeping the hardcoded values?

File: src/status_im/ui/components/connectivity/subs.cljs [hunk 5]
```diff
@@ -36,9 +37,13 @@
                        mailserver-request-error?
                        :t/mailserver-request-error-status
 
+                       (and (= network-type "cellular")
```
Comment (mandrigin) [inline src/status_im/ui/components/connectivity/subs.cljs]: You've defined a function called `cellular?` Maybe we can re-use it  instead of keeping the hardcoded values?

**Review by mandrigin — CHANGES_REQUESTED** [review]
a few tiny changes

File: src/status_im/ui/screens/mobile_network_settings/events.cljs (hunk 3 above)
Comment (rasom) [reply src/status_im/ui/screens/mobile_network_settings/events.cljs:37]: re `remember?`, is it better now or you want it to be something like `:default`, `:true`, `:false`?

File: src/status_im/mailserver/core.cljs (hunk 4 above)
Comment (rasom) [reply src/status_im/mailserver/core.cljs]: fixed

File: src/status_im/ui/screens/mobile_network_settings/events.cljs (hunk 3 above)
Comment (cammellos) [reply src/status_im/ui/screens/mobile_network_settings/events.cljs:37]: I'd rather have it as the same type (`:not-answered :true :false`), rather than :default true false

**Review by mandrigin — APPROVED** [review]

File: src/status_im/ui/screens/mobile_network_settings/events.cljs (hunk 3 above)
Comment (rasom) [reply src/status_im/ui/screens/mobile_network_settings/events.cljs:37]: well, it means that in [some places](https://github.com/status-im/status-react/blob/54d26a723d989341b0d84ce65d7feffbd7f318c3/src/status_im/ui/screens/mobile_network_settings/events.cljs#L75 ) we need to convert boolean to this :true/:false thing, I realize the concern around troolean thing, but I have doubts it is worth doing here

Comment (cammellos) [reply src/status_im/ui/screens/mobile_network_settings/events.cljs:37]: oh I see, never mind then, probably not worth

**Comment by asemiankevich** [comment]
@rasom would you mind to correct a small typo? 

image

**Comment by rasom** [comment]
@asemiankevich done

**Comment by hesterbruikman** [comment]
@rasom @asemiankevich just to be on the safe side. There was some copy in the designs that needed an update: "Until connected to wi-fi?" was this copy updated? Should be "Until connected to Wi-Fi"

**Comment by rasom** [comment]
nope it wasn't, I will file a separate PR for this. Thanks @hesterbruikman
